### PR TITLE
✨ Source Google Analytics 4: make lookback window configurable

### PR DESCRIPTION
**Pull request by gvillafanetapia**
## What

As of version 2.4.2 the lookback window is hardcoded in 2 days.
From our experience the attribution delay can be longer, so its best to leave this for the user to decide.

## How

* Add a new `lookback_window` setting in the spec
* Use this config instead of the hardcoded value, keep the hardcoded 2 as default.

## Review guide
1. `airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/spec.json`
2. `airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py`

## User Impact
* The user will be able to choose a lookback window that suits their business case.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: docs/integrations/sources/google-analytics-data-api.md
Comment (marcosmarxm) [inline docs/integrations/sources/google-analytics-data-api.md:267]: Can you also update the documentation in the connector configuration showing this is configurable now?

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py:50]: Maybe rename this to `DEFAULT_LOOKBACK_WINDOW`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments!

File: docs/integrations/sources/google-analytics-data-api.md
Comment (gvillafanetapia) [reply docs/integrations/sources/google-analytics-data-api.md:267]: sure!

File: airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py
Comment (gvillafanetapia) [reply airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py]: https://github.com/airbytehq/airbyte/pull/38884/commits/9d90f2c2897ef21d332f646fd3fb735a08b34cb0

File: docs/integrations/sources/google-analytics-data-api.md
Comment (gvillafanetapia) [reply docs/integrations/sources/google-analytics-data-api.md]: https://github.com/airbytehq/airbyte/pull/38884/commits/625a1541ae801c122bfd41ec0bcb667c779f2f7a

**Comment by marcosmarxm** [comment]
@gvillafanetapia it looks maintainers doens't have access to your fork/branch. Can you run locally `airbyte-ci format fix all`?

**Comment by marcosmarxm** [comment]
Running tests again @gvillafanetapia !

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
@gvillafanetapia your change looks good to me. Because the change is impacting a common connector I need the approval from another team member to finish and merged it.

File: airbyte-integrations/connectors/source-google-analytics-data-api/sample_files/configured_catalog.json
Comment (maxi297) [inline airbyte-integrations/connectors/source-google-analytics-data-api/sample_files/configured_catalog.json:1]: This file seems identical to `integration_tests/configured_catalog.json` except from the missing `cohort_report` stream. I don't see this file reference anywhere. Why do we need this file? Can we delete it?

The same question applies to `airbyte-integrations/connectors/source-google-analytics-data-api/sample_files/sample_config.json`

**Review by maxi297 — APPROVED** [review]
I'm wondering if we actually need the files in the `sample_files` folder. Other than that, I'm fine with this change. Thanks for your contribution!

**Comment by marcosmarxm** [comment]
@maxi297 I'll remove the files in a separated PR.

**Comment by marcosmarxm** [comment]
Hello @gvillafanetapia your feedback matters a lot to us. Can you spare just 3 minutes to complete a [survey](https://form.typeform.com/to/DOBqejte)? We’re dedicated to making the contributor experience even better, and your input is key to achieving excellence. Thank you for helping us improve!

**Pull request merged**
